test/test_stages.js

```diff
@@ -0,0 +1,396 @@
+        wait(1);
+        try {
+            await privateOffer.sendTransaction({
+                from: buyerAddress,
+                value: web3.toWei(10, 'ether')
+            });
+            assert.ifError("Error, not whitlisted buyers shouldn't be able to buy tokens");
+        } catch (err) {
+            assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Expected rever error after buying tokens as not KYC");
+        }
+        await controller_instance.addBuyerToWhitelist(buyerAddress);
+
+        try {
+            await privateOffer.sendTransaction({
+                from: buyerAddress,
+                value: web3.toWei(9.9, 'ether')
+            });
+            assert.ifError('Error, mininmum amount of ETH at private offer stage is 10 ETH');
+        } catch (err) {
+            assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Expected rever error after sending too small amount of ether");
+        }
+
+        try {
+            await privateOffer.sendTransaction({
+                from: buyerAddress,
+                value: web3.toWei(201, 'ether')
+            });
+            assert.ifError('Error, maximum amount of ETH at private offer stage is 200 ETH');
+        } catch (err) {
+            assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Expected rever error after sending too big amount of ether");
+        }
+
+
+        await privateOffer.sendTransaction({from: buyerAddress, value: web3.toWei(10, 'ether')});
+        let expectedTokenBalance = BigNumber(web3.toWei(120000, 'ether'));
+        let actualTokenBalance = await token.balanceOf(buyerAddress);
+        assert.isTrue(expectedTokenBalance.isEqualTo(actualTokenBalance), "Wrong balance for purchase without bonus");
+
+        await privateOffer.sendTransaction({from: buyerAddress, value: web3.toWei(25, 'ether')});
+        expectedTokenBalance = expectedTokenBalance.plus(BigNumber(web3.toWei(345000, 'ether')));
+        actualTokenBalance = await token.balanceOf(buyerAddress);
+        assert.isTrue(expectedTokenBalance.isEqualTo(actualTokenBalance), "Wrong balance for purchase with 15% bonus");
+
+        await privateOffer.sendTransaction({from: buyerAddress, value: web3.toWei(100, 'ether')});
+        expectedTokenBalance = expectedTokenBalance.plus(BigNumber(web3.toWei(1440000, 'ether')));
+        actualTokenBalance = await token.balanceOf(buyerAddress);
+        assert.isTrue(expectedTokenBalance.isEqualTo(actualTokenBalance), "Wrong balance for purchase with 20% bonus");
+
+        let expectedEthBalance = BigNumber(web3.toWei(135, 'ether'));
+        let actualEthBalance = BigNumber(await web3.eth.getBalance(escrowAddress)).minus(escrowAddressInitialBalance);
+        assert.isTrue(expectedEthBalance.isEqualTo(actualEthBalance), "Wrong Ethereum balance of escrow address");
+
+        let expectedBuyerSpent = BigNumber(0);
```

Comment:
Is the value is zero because we don't track buyer spent amounts during the private offer?
Please add a comment about it.

Reply:
Yes
You are right
People get tokens at private offer couldn't refund